src/QMCDrivers/Optimizers/HybridEngine.cpp

```diff
@@ -4,28 +4,104 @@
 #include <string>
 
 #include "QMCDrivers/Optimizers/HybridEngine.h"
+#include "OhmmsData/ParameterSet.h"
 #include "Message/CommOperators.h"
+#include "OhmmsData/XMLParsingString.h"
 
 
 namespace qmcplusplus
 {
-HybridEngine::HybridEngine(Communicate* comm, const xmlNodePtr cur) : myComm(comm)
+HybridEngine::HybridEngine(Communicate* comm, const xmlNodePtr cur) : myComm(comm) { processXML(cur); }
+
+
+bool HybridEngine::processXML(const xmlNodePtr opt_xml)
 {
-  //Number of steps in a descent section of hybrid method
-  m_param.add(descent_len, "descent_length", "int");
-  //Number of steps in a BLM of section of hybrid method
-  m_param.add(blm_len, "BLM_length", "int");
-  processXML(cur);
+  opt_methods_.clear();
+  saved_xml_opt_methods_.clear();
+  num_updates_opt_methods_.clear();
+
+  xmlNodePtr cur = opt_xml->children;
+  while (cur != NULL)
+  {
+    std::string cname((const char*)(cur->name));
+    if (cname == "optimizer")
+    {
+      std::string children_MinMethod;
+      ParameterSet m_param;
+      m_param.add(children_MinMethod, "MinMethod", "string");
+      m_param.put(cur);
+
+      if (children_MinMethod.empty())
+        throw std::runtime_error("MinMethod must be given!\n");
+      XMLAttrString updates_string(cur, "num_updates");
+      app_log() << "HybridEngine saved MinMethod " << children_MinMethod << " num_updates = " << updates_string
+                << std::endl;
+      auto iter = OptimizerNames.find(children_MinMethod);
+      if (iter == OptimizerNames.end())
+        throw std::runtime_error("Unknown MinMethod!\n");
+      opt_methods_.push_back(iter->second);
+      saved_xml_opt_methods_.push_back(cur);
+      num_updates_opt_methods_.push_back(std::stoi(updates_string));
+    }
+    cur = cur->next;
+  }
+
+  if (saved_xml_opt_methods_.size() != 2)
+    throw std::runtime_error("MinMethod hybrid needs two optimizer input blocks!\n");
+
+  return true;
 }
 
+xmlNodePtr HybridEngine::getSelectedXML(int counter) const
+{
+  return saved_xml_opt_methods_[identifyMethodIndex(counter)];
+}
 
-bool HybridEngine::processXML(const xmlNodePtr cur)
+bool HybridEngine::queryStore(int counter, int store_num, const OptimizerType method) const
 {
-  m_param.put(cur);
+  bool store = false;
 
+  int idx = 0;
+  for (int i = 0; i < opt_methods_.size(); i++)
+  {
+    if (opt_methods_[i] == method)
```

Comment:
Please replace this look up by std::find method in the opt_methods_.
define the case if not found.